scripts/exec.mjs

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-void, no-console, functional/no-expression-statement, jsdoc/require-description, valid-jsdoc */
+import { spawn } from 'cross-spawn';
+
+/**
+ * Async exec.
+ *
+ * Inspired by `remix-run` build scripts, [reference](https://github.com/remix-run/remix/blob/534e1ec071f17654a4db8622e30d6ff70548ce26/scripts/build.mjs).
+ *
+ * @param {string} command - Shell command to execute.
+ * @param {string[]} [args] - Command arguments.
+ * @returns {Promise} Returns prom.
+ */
+export function exec(command, args) {
+  /** @type {(data: Error) => any} */
+  const handleError = (data) => console.error(data.toString().trim());
+
+  return new Promise((resolve, reject) => {
+    const child = spawn(command, args, {
+      cwd: process.cwd(),
+      stdio: 'inherit',
+    });
+
+    const onExit = () => child.kill('SIGINT');
+
+    process.on('SIGINT', onExit);
+    process.on('SIGTERM', onExit);
+
+    child.on('error', handleError);
+    child.on('close', (code) => {
+      if (code === 0) {
+        resolve(void 0);
+      } else {
+        reject(new Error(`${command} exited with code ${code}`));
+      }
+    });
+  });
+}
```

Comment:
Only things I did differently than remix was to inherit the io from the parent process, and also exit the child if the parent does. Otherwise if you try to cancel the build process with `^C` the child will keep running.